### ISSUE TITLE
UserInfoJson.attributes deserialization during OIDC authentication

### ISSUE THREAD

**Issue by Astimus** [grafana/grafana#41883]
I use Grafana with oidc authentication. For some users my userinfo endpoint returns json with attributes property in the format:
`"attributes": "{\"propety\":[\"propertyValue\"]}" `
Grafana fails with the deserialization of this property with error: 
`cannot unmarshal string into Go struct field UserInfoJson.attributes of type map[string][]string`

I want Grafana to ignore this property in my case. Is there any protocol standard which says that attributes property must be in format map[string][]string ? I haven't found anything like that.

**Environment**:
- Grafana v7.2.0-pre
- IdentityServer4 OIDC authentication

**Comment by grafanabot** [comment]
Please ask your question on [community.grafana.com/](https://community.grafana.com/). To avoid having your issue closed in the future, please read our [CONTRIBUTING](https://github.com/grafana/grafana/blob/main/CONTRIBUTING.md) guidelines.

Happy graphing!

**Comment by kkalev** [comment]
We are experiencing the same problem described above. We have integrated Grafana with an SSO./OIDC authentication server, specifically Apereo [CAS](https://github.com/apereo/cas). The attributes returned by the authentication server have the following structure:
```
"attributes": {
    "property": [
        "propertyValue1",
        "propertyValue2"
    ],
    "property": "propertyValue"
}
```
In the case where there is only one propertyValue, CAS returns it as a string instead of an array of strings, which is causing compatibility issues with Grafana. Grafana expects attributes to be in the format map[string][]string as defined [here](https://github.com/grafana/grafana/blob/main/pkg/login/social/generic_oauth.go#LL86C53-L86C53), but CAS does not adhere to this format when there is a single value.

We have explored the CAS documentation and did not find any specific protocol standard that mandates the attributes property to be in the map[string][]string format. Therefore, it seems that the attribute format is not standardized across different OIDC providers.

We believe it would be beneficial to have a workaround or configuration option in Grafana that allows it to handle attributes with varying value types, accommodating cases where a single value is returned as a string. This would help ensure compatibility between Grafana and authentication servers that do not conform to the map[string][]string format for attributes.

Is there any guidance or suggested workaround that can be provided to make these two software solutions compatible with each other? Apereo CAS is one of the most widely used SSO solutions in the market (especially open source), so we are sure this is relevant to many other users of Grafana.
